### ISSUE TITLE
NERVE problems, again...

### ISSUE THREAD

**Issue by Zodomaniac** [fabiangreffrath/crispy-doom#269]
Version of Crispy Doom: Daily build 13.02.2018

Operating System and version: Windows 7

Game: (Doom/Heretic/Hexen/Strife/other): Doom 2 (No Rest for the Living)

Any loaded WADs and mods (please include full command line): `start crispy-doom -iwad doom2(2).wad -merge DoomMetalVol4_44100.wad -loadgame 2 -playdemo nerve_test_secret_exit`

### Bug description

Observed behavior: When recording a demo, the secret exit from MAP04: Hell Mountain takes me to MAP01: The Earth Base instead of MAP09: March of the Demons.
[Nerve secret demo test.zip](https://github.com/fabiangreffrath/crispy-doom/files/1720535/Nerve.secret.demo.test.zip)

In turn, when recording a demo, MAP09: March of the Demons follows MAP08: Tomb of Malevolence, and when I finish March of the Demons while recording the demo, the game crashes without even writing the demo file!

Expected behavior: The secret exit from MAP04: Hell Mountain should take the player to MAP09: March of the Demons which should in turn exit to MAP05: Vivisection.

**Comment by fabiangreffrath** [comment]
> When recording a demo, the secret exit from MAP04: Hell Mountain takes me to MAP01: The Earth Base instead of MAP09: March of the Demons.

That's what Vanilla and PrBoom+ would do.

> when recording a demo, MAP09: March of the Demons follows MAP08: Tomb of Malevolence

Again, that's what Vanilla and PrBoom+ would do.

> when I finish March of the Demons while recording the demo, the game crashes without even writing the demo file

This, in turn, shouldn't happen!

**Comment by fabiangreffrath** [comment]
> when I finish March of the Demons while recording the demo, the game crashes without even writing the demo file

Hm, I cannot reproduce this. When I finish NERVE's MAP09 while recording a demo, it takes me to MAP10 of DOOM2.WAD, just as expected.

**Comment by Zodomaniac** [comment]
Could you record a demo with NERVE MAP09+start of MAP10, so that I could see if it plays back?

**Comment by Zodomaniac** [comment]
Maybe the matter is that my build is 5.1 with EXEs from the daily build?

**Comment by fabiangreffrath** [comment]
Na, I am always using the latest development code.

**Comment by Zodomaniac** [comment]
Used the latest dev build, the crash doesn't occur if I `-merge nerve.wad`, but it does if this command is not present.

**Comment by fabiangreffrath** [comment]
But, how are you supposed you play back a NERVE demo without adding the PWAD on the command line at all? Could you please send me this demo so I can investigate?

**Comment by Zodomaniac** [comment]
Here you are!
I had to record the demo with merging NERVE.WAD (otherwise it crashes with level transition).
To play back the demo, I included 2 BAT-files, one with -merge and the other without it.
In the first case the level transition works, in the other the demo runs (i.e. the level is detected correctly from the demo) but crashes the program on level transition.
[nerve mydemo.zip](https://github.com/fabiangreffrath/crispy-doom/files/1731227/nerve.mydemo.zip)
